### PR TITLE
Support for SystemD

### PR DESCRIPTION
**Pull request by shaneramey**
## Description
Support for SystemD

## Motivation and Context
This enables support for SystemD in the sensu_service LWRP

## How Has This Been Tested?
Tested in Test-Kitchen

## Screenshots (if appropriate):
none
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by runningman84** [comment]
Can this be merged soon?

File: .kitchen.yml [hunk 1]
```diff
@@ -64,6 +64,13 @@ suites:
       sensu:
         rabbitmq:
           user: sensu_client
+  - name: systemd
+    run_list:
+      - recipe[sensu-test]
+    excludes:
```
Comment (tas50) [inline .kitchen.yml]: Using includes would make a lot more sense hete

Comment (tas50) [inline .kitchen.yml]: Here*

**Comment by cwjohnston** [comment]
Hi @shaneramey, thanks for the pull request.

I'm still learning about systemd myself, but it seems like your proposed implementation relies on `systemd-sysv-generator` to generate unit files that wrap the default sysv init scripts. In my testing so far it doesn't seem this will be sufficient.

When I test your changes in test-kitchen on ubuntu-1404 and ubuntu-1204, the services run but I'm pretty sure the services are actually being managed by upstart. e.g. there's no systemctl available in root's path, but the systemd-shim package is present to help us "fake it".

I also tested against the ubuntu-1604 platform where I'm sure systemd is being used, but systemctl thinks service isn't running:

```
vagrant@systemd-ubuntu-1604:~# systemctl status sensu-client
* sensu-client.service - LSB: Sensu monitoring framework client
   Loaded: loaded (/etc/init.d/sensu-client; bad; vendor preset: enabled)
   Active: inactive (dead)
     Docs: man:systemd-sysv-generator(8)
```

I think this is because Sensu services [are forking into the background](http://unix.stackexchange.com/questions/137028/why-is-my-systemd-unit-loaded-but-inactive-dead), and that if we want the unit files generated by `systemd-sysv-generator` to work, they will need to include `PIDFile` directives pointed at the PID files in `/var/run/sensu`.

I suspect that a fully functional implementation will need to make use of the systemd unit files that ship with Sensu (see /usr/share/sensu/systemd or https://github.com/sensu/sensu-build/tree/master/sensu_configs/systemd on github). Have you had a look at the options documented here? Any thoughts?

It would be great if this suite also had tests which verify that systemctl sees the service is running, etc. so that we know when a platform isn't really managing one or more Sensu services under systemd.

**Comment by willejs** [comment]
@cwjohnston Im also looking at this, im going to have a go now, and make another PR using those as templates

**Comment by cwjohnston** [comment]
Marking this as wontfix for now, as we're working on refactoring our service scripts to better support systemd and others for Sensu 0.27 so this will all change soon anyway.

**Pull request closed without merging**